### PR TITLE
[RealSense] Update to 2.54.1

### PR DESCRIPTION
**Pull request by Tamir91**
RealSense version updated to 2.54.1
OSX patch removed

### CODE REVIEW

**Comment by Tamir91** [comment]
> 

@microsoft-github-policy-service agree company="Intel"

**Comment by MonicaLiu0311** [comment]
```
##[error]Found the following errors:
##[error]versions/r-/realsense2.json(2,): error : while parsing versions for realsense2 from version/r-/realsense2.json
2.54.1 is declared with d8884a0ab29863bd7fa8aeb6167c75a8b66513dc, but the local port has a different SHA cb6ef9deafcbb447cbb7903412b4a1acca8b20ea.
```
Please run `./vcpkg x-add-version portName --overwrite-version`.

`--overwrite-version`
After the first modification, please run `./vcpkg x-add-version portName` to generate the corresponding json; for subsequent modifications of the same version, please use `./vcpkg x-add-version portName --overwrite-version` to update git-tree.

**Comment by Tamir91** [comment]
> ```
> ##[error]Found the following errors:
> ##[error]versions/r-/realsense2.json(2,): error : while parsing versions for realsense2 from version/r-/realsense2.json
> 2.54.1 is declared with d8884a0ab29863bd7fa8aeb6167c75a8b66513dc, but the local port has a different SHA cb6ef9deafcbb447cbb7903412b4a1acca8b20ea.
> ```
> 
> Please run `./vcpkg x-add-version portName --overwrite-version`.
> 
> `--overwrite-version` After the first modification, please run `./vcpkg x-add-version portName` to generate the corresponding json; for subsequent modifications of the same version, please use `./vcpkg x-add-version portName --overwrite-version` to update git-tree.

Hi, Monica!

Thank you for your aid!
I generated new git-tree key...

**Comment by MonicaLiu0311** [comment]
An error occurred while testing usage.


usage:
```
realsense2 provides CMake targets:

    # this is heuristically generated, and may not be correct
    find_package(realsense2 CONFIG REQUIRED)
    target_link_libraries(main PRIVATE realsense2::fw realsense2::rsutils realsense2::realsense2 realsense2::realsense-file)
```

When testing usage, the following error occurs:
```
>------ Rebuild All started: Project: CMakeFindUsage, Configuration: x64-Debug ------
  MSBuild version 17.6.3+07e294721 for .NET Framework
  
  MSBuild version 17.6.3+07e294721 for .NET Framework
  
    Checking Build System
    Building Custom Rule C:/Users/monica/source/repos/CMakeFindUsage/CMakeFindUsage/CMakeLists.txt
    CMakeFindUsage.cpp
C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.36.32532\include\ostream(774,1): warning C4530: C++ exception handler used, but unwind semantics are not enabled. Specify /EHsc 
  C:\Users\monica\source\repos\CMakeFindUsage\CMakeFindUsage\CMakeFindUsage.cpp(10,7): message : see reference to function template instantiation 'std::basic_ostream<char,std::char_traits<char>> &std::operator <<<std::char_traits<char>>(std::basic_ostream<char,std::char_traits<char>> &,const char *)' being compiled [C:\Users\monica\source\repos\CMakeFindUsage\out\build\x64-Debug\CMakeFindUsage\main.vcxproj]
C:\Users\monica\source\repos\CMakeFindUsage\out\build\x64-Debug\CMakeFindUsage\LINK : fatal error LNK1104: cannot open file 'E:\realsense\installed\x64-windows\common\fw\fw.res' 

Rebuild All failed.
```

<details>
<summary><b>CMakeFindUsage.cpp</b></summary>
<pre>
#include &#60;iostream&#62;
#include "librealsense2/rs.h"

using namespace std;

int main()
{
        cout << "Hello CMake." << endl;
        return 0;
}
</pre></details>


<details>
<summary><b>CMakeLists.txt</b></summary>
<pre>
cmake_minimum_required (VERSION 3.8)

set(CMAKE_TOOLCHAIN_FILE "E:/realsense/scripts/buildsystems/vcpkg.cmake")

project ("CMakeFindUsage")

add_executable (main "CMakeFindUsage.cpp")

find_package(realsense2 CONFIG REQUIRED)

target_link_libraries(main PRIVATE realsense2::fw realsense2::rsutils realsense2::realsense2 realsense2::realsense-file)
</pre></details>

Upstream related: https://github.com/IntelRealSense/librealsense/blob/v2.54.1/common/fw/CMakeLists.txt#L41-L47.

**Comment by MonicaLiu0311** [comment]
Currently this file is in `E:\realsense\buildtrees\realsense2\x64-windows-rel\common\fw\fw.res`, you can use the following statement in `portfile.cmake` to install it to the specified location `E:\realsense\installed\ x64-windows\common\fw\fw.res`:


```
if(VCPKG_TARGET_IS_WINDOWS)
    ...
    file(MAKE_DIRECTORY "${CURRENT_PACKAGES_DIR}/common/fw")
    file(INSTALL "${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}-rel/common/fw/fw.res" DESTINATION "${CURRENT_PACKAGES_DIR}/common/fw")
endif()
```

**Comment by Tamir91** [comment]
> Currently this file is in `vcpkg\buildtrees\realsense2\x64-windows-rel\common\fw\fw.res`, you can use the following statement in `portfile.cmake` to install it to the specified location `vcpkg\installed\ x64-windows\common\fw\fw.res`:
> 
> ```
> if(VCPKG_TARGET_IS_WINDOWS)
>     ...
>     file(MAKE_DIRECTORY "${CURRENT_PACKAGES_DIR}/common/fw")
>     file(INSTALL "${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}-rel/common/fw/fw.res" DESTINATION "${CURRENT_PACKAGES_DIR}/common/fw")
> endif()
> ```

Hi, Monica!

I added lines according your guidelines. I will push it now.

Thank you for your support!

**Comment by MonicaLiu0311** [comment]
All features are tested successfully in the following triplet:
```
x86-windows
x64-windows
```

**Review by MonicaLiu0311 — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thanks!

**Comment by Tamir91** [comment]
Hi, Monica

I noted that the RealSense version was changed to 2.53.1 but it still an old version...
May you explain about this repo behavior? 

image
 Thank you!

**Comment by MonicaLiu0311** [comment]
> Hi, Monica
> 
> I noted that the RealSense version was changed to 2.53.1 but it still an old version... May you explain about this repo behavior?
> 
> image Thank you!

Because it is not yet able to maintain real-time synchronization with vcpkg.
See related issues: https://github.com/vcpkg/vcpkg.github.io/issues/48
See related PR: https://github.com/vcpkg/vcpkg.github.io/pull/60

File: ports/realsense2/vcpkg.json [hunk 1]
```diff
@@ -25,9 +25,6 @@
         "openni2"
       ]
     },
-    "tm2": {
-      "description": "Build Intel® RealSense™ T265 device"
-    },
```
Comment (autoantwort) [inline ports/realsense2/vcpkg.json:30]: It seems that `rtabmap` still depends on this:

https://github.com/microsoft/vcpkg/blob/ad46340bfce415333d6a2139592c22a499fb0df0/ports/rtabmap/vcpkg.json#L117-L130

**Pull request merged**